tests/mesh/boundary_info.C

```diff
@@ -770,6 +816,23 @@ public:
         CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(2), container.size());
         CPPUNIT_ASSERT_EQUAL(static_cast<boundary_id_type>(5), container[0]);
         CPPUNIT_ASSERT_EQUAL(static_cast<boundary_id_type>(3), container[1]);
+
+        std::vector<std::vector<boundary_id_type>> all_bids_container;
+        bi.side_boundary_ids(elem, all_bids_container);
+        CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(elem->n_sides()), all_bids_container.size());
+        CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(2), all_bids_container[3].size());
+        CPPUNIT_ASSERT_EQUAL(static_cast<boundary_id_type>(5), all_bids_container[3][0]);
+        CPPUNIT_ASSERT_EQUAL(static_cast<boundary_id_type>(3), all_bids_container[3][1]);
+        CPPUNIT_ASSERT_EQUAL(static_cast<std::size_t>(1), all_bids_container[2].size());
+        CPPUNIT_ASSERT_EQUAL(static_cast<boundary_id_type>(2), all_bids_container[2][0]);
+        // This is against the spirit of the test, but harmless
```

Comment:
Huh.  I actually disagree that this is harmless, I think you've revealed a bug in our spec.  It's kind of a time bomb to allow this behavior.  Setting `allow_children_on_boundary_side(false)` should probably be actively doing a `transfer_boundary_ids_from_children()` and then deleting the children's data, not leaving the children's data there-but-ignored.

This PR isn't the right place to fix any of that, though; I'd even leave this test here, just update the comment to note that it's undefined behavior and will probably be changed later.